# TESTING OF HYPOTHESIS

The question is:

A company started to invest in digital marketing as a new way of their product promotions. For that they collected data and decided to carry out a study on it.

● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

● The company needs to check whether there is any dependency between the features “Region” and “Manager”.

Help the company to carry out their study with the help of data provided.



The given dataset consists of 22 rows and 5 columns namely Month, Region, Manager, Sales_before_digital_add(in $) and

Sales_After_digital_add(in $).



In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import scipy.stats as stats

In [2]:
data=pd.read_csv('D:/python programming/Sales_add.csv')
data

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [3]:
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [4]:
data.shape

(22, 5)

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [6]:
data.isna().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

Hence, no null values present in the data

# 1. Whether there is any increase in sales after stepping into digital marketing

In [7]:
data[['Sales_before_digital_add(in $)','Sales_After_digital_add(in $)']].describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

We have to check whether there is any increase in sales after stepping into digital marketing . For that we should carry out one-tailed t-test such that

NULL HYPOTHESIS: Average sales before and after digital marketing are same against

ALTERNATIVE HYPOTHESIS:Average sales after digital marketing is greater than before digital marketing

Next, consider the p-value,

If the p-value is less than your significance level, you can reject the null hypothesis and conclude that the effect is statistically significant. In other words, the evidence in your sample is strong enough to be able to reject the null hypothesis at the population level.

Here, the significane level is assumed to be 0.05.



In [8]:
ttest,pval = stats.ttest_rel(data['Sales_before_digital_add(in $)'],data['Sales_After_digital_add(in $)'])
print(pval)

6.336667004575778e-11


INSIGHT: Here, the p-value, 6.336667004575778e-11, is less than the significance level, 0.05.

Hence, we should reject the null hypothesis and so accept alternative hypothesis.

So, we can conclude that the average sales after digital marketing is greater than before digital marketing.

Hence, there is an increase in sales after stepping into digital marketing.

# 2.Whether there is any dependency between the features “Region” and “Manager”2.

We have to check whether there is any dependency between the features “Region” and “Manager”. For that we should carry out chi-square test for independence such that

NULL HYPOTHESIS:There is dependency between the features “Region” and “Manager” against

ALTERNATIVE HYPOTHESIS:There is not any dependency between the features “Region” and “Manager”



In [9]:
data_chi = pd.read_csv('D:/python programming/Sales_add.csv')
contingency_table=pd.crosstab(data_chi["Region"],data_chi["Manager"])
print('contingency_table :-\n',contingency_table)

contingency_table :-
 Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1


In [10]:
#Observed Values
Observed_Values = contingency_table.values 
print("Observed Values :-\n",Observed_Values)

Observed Values :-
 [[4 3 3]
 [4 1 2]
 [1 3 1]]


In [11]:
b=stats.chi2_contingency(contingency_table)
Expected_Values = b[3]
print("Expected Values :-\n",Expected_Values)

Expected Values :-
 [[4.09090909 3.18181818 2.72727273]
 [2.86363636 2.22727273 1.90909091]
 [2.04545455 1.59090909 1.36363636]]


In [12]:
no_of_rows=len(contingency_table.iloc[0:3,0])
no_of_columns=len(contingency_table.iloc[0,0:3])
ddof=(no_of_rows-1)*(no_of_columns-1)
print("Degree of Freedom:-",ddof)
alpha = 0.05

Degree of Freedom:- 4


In [13]:
from scipy.stats import chi2
chi_square=sum([(o-e)**2./e for o,e in zip(Observed_Values,Expected_Values)])
chi_square_statistic=chi_square[0]+chi_square[1]
print("chi-square statistic:-",chi_square_statistic)

chi-square statistic:- 2.921995464852608


In [14]:
critical_value=chi2.ppf(q=1-alpha,df=ddof)
print('critical_value:',critical_value)

critical_value: 9.487729036781154


In [15]:
#p-value
p_value=1-chi2.cdf(x=chi_square_statistic,df=ddof)
print('p-value:',p_value)

p-value: 0.5709629929220089


In [16]:
print('Significance level: ',alpha)
print('Degree of Freedom: ',ddof)
print('chi-square statistic:',chi_square_statistic)
print('critical_value:',critical_value)
print('p-value:',p_value)

Significance level:  0.05
Degree of Freedom:  4
chi-square statistic: 2.921995464852608
critical_value: 9.487729036781154
p-value: 0.5709629929220089


if chi-square_statistic >=critical_value: Reject null hypothesis, ie; There is not any relationship between 2 categorical variables else: Accept null hypothesis, ie; There is a relationship between 2 categorical variables

if p_value <=alpha: Reject null hypothesis, ie; There is not any relationship between 2 categorical variables else: Accept null hypothesis, ie; There is a relationship between 2 categorical variables.

INSIGHT:- Here, the chi-square statistic, 2.921995464852608, is less than the critical value, 9.487729036781154 , so, we accept the null hypothesis

OR

We can consider the p-value, 0.5709629929220089, which is greater than the significance level, 0.05, we accept the null hypothesis.

Hence, we can conclude that there is a dependency between the features “Region” and “Manager”.

ie; there is an association between “Region” and “Manager”.